extension/src/install.ts

```diff
@@ -89,18 +92,36 @@ export async function updateExtensions(extensionInfo: ExtensionInfoService, pack
         await showReloadPrompt(
             localize(
                 'reload.to.complete.update.all',
-                'Please reload Visual Studio Code to complete updating the extensions.',
+                'to complete updating the extensions.',
             ),
         );
     }
 }
 
 /**
  * Displays a message with a button to reload vscode.
- * @param message The message to display.
+ * @param suffixMessage The reason to reload VSCode. The message will be displayed in a "popin".
```

Comment:
This will work for English, but I don't know if it will work for every language. Here are a couple ideas I came up with:

Idea 1: undo the changes to this parameter and just use a generic message when auto reloading:

```ts
export async function showReloadPrompt(message: string): Promise<void> {
    if (getAutoReload()) {
        message = localize(
            'automatic.reload',
            'Visual Studio Code will restart in 3 seconds to complete changes to private extensions.'
        );
        // ...
    } else {
        reload = await vscode.window.showInformationMessage(message, localize('reload.now', 'Reload Now'));
    }
```

Idea 2: pass in enough information for showReloadPrompt() to build the entire message.
``` ts
enum ReloadReason {
    Install,
    Uninstall,
    Update,
    UpdateAll,
}

function getReloadPromptMessage(reason: ReloadReason, extension?: string) {
    switch (reason) {
        case ReloadReason.Install:
            return localize(
                'reload.to.complete.install',
                'Please reload Visual Studio Code to complete installing the extension {0}.',
                extension,
            );
        // ...
    }
}

function getAutoReloadMessage(reason: ReloadReason, extension?: string) {
    switch (reason) {
        case ReloadReason.Install:
            return localize(
                'reload.to.complete.install',
                'Visual Studio Code will restart in 3 seconds to complete installing the extension {0}.',
                extension,
            );
        // ...
    }
}

/**
 * Displays a message with a button to reload vscode.
 * @param message The message to display.
 * @param name of the extension that was changed.
 */
export async function showReloadPrompt(reason: ReloadReason, extension?: string): Promise<void> {
    if (getAutoReload()) {
        const message = getAutoReloadMessage(reason, extension);
        // ...
    } else {
        const message = getReloadPromptMessage(reason, extension);
        // ...
    }
```

Reply:
I keep the second idea :)